From d6a686729a1ed59ee57ed6f625c74f46c7f02fae Mon Sep 17 00:00:00 2001
From: Luca Capello <luca@pca.it>
Date: Thu, 4 Oct 2012 12:04:23 +0200
Subject: [PATCH] yodack: add keyring option

The Debian Maintainer keyring shipped in the debian-keyring package is
not always in sync with the real situation, i.e. the list available at
<http://ftp-master.debian.org/dm-uploaders.html>.
---
 man1/yodack.1 |    7 ++++---
 yodack        |    9 +++++++--
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/man1/yodack.1 b/man1/yodack.1
index 08a4ffa..dfe3cf3 100644
--- a/man1/yodack.1
+++ b/man1/yodack.1
@@ -4,9 +4,9 @@
 .SH "NAME"
 yodack \- Ye Olde Debian Archive Control Kit
 .SH "SYNOPSIS"
-.BI "yodack " ftp\-master " dm " example@org " allow " "this that" " deny " something\-else
+.BI "yodack " ftp\-master " dm " example@org " keyring " keyring.gpg " allow " "this that" " deny " something\-else
 
-.BI "yodack at " ftp\-master " for dm " example@org " allow upload of package " "this, that" ", but deny uploading " something\-else.
+.BI "yodack at " ftp\-master " for dm " example@org " in keyring " keyring.gpg " allow upload of package " "this, that" ", but deny uploading " something\-else.
 
 .SH "DESCRIPTION"
 Ye Olde Debian Archive Control Kit (or \fByodack\fR for short) is a
@@ -36,7 +36,8 @@ end up being sent, however.
 Grant or revoke access to a particular Debian Maintainer (identified
 by the \fIkey\-id\fR) on the listed packages.
 
-The maintainer will be looked up in the Debian Maintainer keyring, and
+The maintainer will be looked up in the Debian Maintainer keyring by default
+if an alternative keyring is specified via the \fBkeyring\fR option, and
 Yodack will ask for verification - or complain, if there is more than
 one match. Yodack does not like to choose, it wants the young
 apprentices to make their own choices.
diff --git a/yodack b/yodack
index 836c41f..8a6d91a 100755
--- a/yodack
+++ b/yodack
@@ -107,7 +107,7 @@ EOF
 
 yodack_add_dm_fingerprint () {
         if fprs=$(gpg --no-options --no-default-keyring \
-                      --keyring /usr/share/keyrings/debian-maintainers.gpg \
+                      --keyring "$2" \
                       --list-options no-show-photo --fingerprint "$1" 2>/dev/null); then
                 :
         else
@@ -224,6 +224,7 @@ yodack_parse_argv_action_dm () {
         stop=0
         allow=0
         deny=0
+        keyring=/usr/share/keyrings/debian-maintainers.gpg
 
         while [ $# -gt 0 ] && [ "$stop" -eq 0 ]; do
                 case "$1" in
@@ -242,6 +243,10 @@ yodack_parse_argv_action_dm () {
                                 deny=1
                                 allow=0
                                 ;;
+                        "keyring")
+                                keyring="$2"
+                                shift
+                                ;;
                         *)
                                 if [ "$allow" -eq 1 ]; then
                                         allow_list="${allow_list} $(yodack_arg_strip $1)"
@@ -256,7 +261,7 @@ yodack_parse_argv_action_dm () {
         done
 
         yodack_add_action "dm"
-        yodack_add_dm_fingerprint "${keyid}"
+        yodack_add_dm_fingerprint "${keyid}" "${keyring}"
         yodack_add_dm_allow "${allow_list}"
         yodack_add_dm_deny "${deny_list}"
         yodack_add_section_end
-- 
1.7.10.4

